src/main/java/dev/cdevents/CDEventTypes.java

```diff
@@ -14,217 +18,922 @@ private CDEventTypes() {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("environment",  environment.toString())
+                        .withExtension("artifactid", artifactId)
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery service upgraded event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param environment
+     * @param artifactId
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * service upgraded event
+     */
+
+    public static CloudEvent createServiceUpgradedEvent(
+            final String eventType, final String id,
+            final Environment environment, final URI artifactId,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("environment", environment.toString())
+                        .withExtension("artifactid", artifactId)
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery service rolled back event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param environment
+     * @param artifactId
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * service rolled back event
+     */
+
+    public static CloudEvent createServiceRolledBackEvent(
+            final String eventType, final String id,
+            final Environment environment, final URI artifactId,
+            final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("environment", environment.toString())
+                        .withExtension("artifactid", artifactId)
+                        .build();
+        return cdToSend;
+    }
+
+    /**
+     * Creates a continuous delivery service removed event
+     * using {@link CloudEventBuilder}
+     * and returns {@link CloudEvent} object.
+     *
+     * @param eventType
+     * @param id
+     * @param environment
+     * @param data
+     * @return the cloudEvent object with continuous delivery
+     * service removed event
+     */
+
+    public static CloudEvent createServiceRemovedEvent(
+            final String eventType, final String id,
+            final Environment environment, final String data) {
+        CloudEvent cdToSend =
+                buildCloudEvent(eventType, data)
+                        .withExtension("id", id)
+                        .withExtension("environment", environment.toString())

```

Comment:
please use jackson-databind to write this `environment` as String, say objectMapper.writeValueAsString(environment)